File: socha-sdk/src/server-api/sc/networking/clients/IHistoryListener.java
package sc.networking.clients;

import sc.api.plugins.IGameState;
import sc.protocol.responses.ProtocolErrorMessage;
import sc.shared.GameResult;

public interface IHistoryListener {
  void onNewState(String roomId, IGameState o);

  void onGameOver(String roomId, GameResult o);

  void onGameError(String roomId, ProtocolErrorMessage error);
}

File: sdk/src/server-api/sc/networking/clients/IHistoryListener.kt
package sc.networking.clients

import sc.api.plugins.IGameState
import sc.protocol.room.ErrorMessage
import sc.shared.GameResult

interface IHistoryListener {
    fun onNewState(roomId: String, state: IGameState)
    fun onGameOver(roomId: String, result: GameResult) {}
    fun onGameError(roomId: String, error: ErrorMessage) {}
}